### PR TITLE
Use input version referenced in RAO request for running tasks instead of the latest arrived version

### PR DESCRIPTION
**Pull request by vbochetRTE**
**Please check if the PR fulfills these requirements** *(please use `'[x]'` to check the checkboxes, or submit the PR and then click the checkboxes)*
- [ ] The commit message follows our guidelines
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**Does this PR already have an issue describing the problem ?** *If so, link to this issue using `'#XXX'` and skip the rest*

**What kind of change does this PR introduce?** *(Bug fix, feature, docs update, ...)*

**What is the current behavior?** *(You can also link to an open issue here)*

**What is the new behavior (if this is a feature change)?**

**Does this PR introduce a breaking change?** *(What changes might users need to make in their application due to this PR?)*

**Other information**:

(if any of the questions/checkboxes don't apply, please delete them entirely)

### CODE REVIEW

File: src/main/java/com/farao_community/farao/core_cc/adapter/service/CoreCCAdapterService.java [hunk 1]
```diff
@@ -0,0 +1,190 @@
...
+                .toList();
+    }
+
+    private void addProcessFileInInputFilesMap(ProcessFileDto processFileDto, EnumMap<FileType, CoreCCFileResource> inputFiles) {
+        final String fileType = processFileDto.getFileType();
+        LOGGER.info("Received {} with DocumentId {}", fileType, processFileDto.getDocumentId());
+        switch (fileType) {
+            case "CGM" -> inputFiles.put(FileType.CGM, getCoreCCFileResource(processFileDto));
+            case "DCCGM" -> {
+                if (null == processFileDto.getFilePath()) {
```
[the remark this answers is not in the document]
Comment (vbochetRTE) [reply src/main/java/com/farao_community/farao/core_cc/adapter/service/CoreCCAdapterService.java:163]: You're right, I made a mistake here when rewriting code and importing changes from Daniel's PR about DCCGM. I will fix it in a future commit.

File: src/main/java/com/farao_community/farao/core_cc/adapter/service/JobLauncherAutoService.java [hunk 2]
```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright (c) 2021, RTE (http://www.rte-france.com)
```
Comment (kahyami) [inline src/main/java/com/farao_community/farao/core_cc/adapter/service/JobLauncherAutoService.java:2]: 2024

File: src/test/java/com/farao_community/farao/core_cc/adapter/service/FileImporterTest.java [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+
+import com.farao_community.farao.core_cc.adapter.exception.RaoRequestImportException;
+import com.farao_community.farao.gridcapa_core_cc.api.resource.CoreCCFileResource;
+import org.assertj.core.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mockito;
+
+import javax.xml.bind.JAXBException;
+
```
Comment (kahyami) [inline src/test/java/com/farao_community/farao/core_cc/adapter/service/FileImporterTest.java:17]: Add author

**Comment by kahyami** [comment]
Please add author and copyright in new classes 
/*
 * Copyright (c) 2024, RTE (http://www.rte-france.com)
 * This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0/.
 */

**Pull request merged**